File: src/main/java/org/manapart/jumpPads/JumpPad.java
package org.manapart.jumpPads;

import net.minecraft.block.*;
import net.minecraft.block.material.Material;
import net.minecraft.block.material.MaterialColor;
import net.minecraft.entity.Entity;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.IBlockReader;
import net.minecraft.world.World;
import net.minecraftforge.common.ToolType;

import javax.annotation.Nullable;

public class JumpPad extends SlimeBlock {
    private final Float velocity;

    JumpPad(Float velocity) {
        super(getProps());
//        super(Block.Properties.create(Material.CORAL, MaterialColor.COLOR_GREEN).hardnessAndResistance(4f).sound(SoundType.SLIME_BLOCK));
        this.velocity = velocity;
    }

    private static AbstractBlock.Properties getProps() {
        Material padMat = new Material.Builder(MaterialColor.COLOR_GREEN).build();
        AbstractBlock.Properties props = AbstractBlock.Properties.of(padMat);
        props.sound(SoundType.SLIME_BLOCK);
        props.strength(4);
        return props;
    }

    @Override
    public void stepOn(World worldIn, BlockPos pos, Entity entityIn) {
        entityIn.fallDistance = 0;
        super.stepOn(worldIn, pos, entityIn);
        if (!entityIn.isCrouching() && isInMiddleOfBlock(pos, entityIn)) {
            if (worldIn.isClientSide) {
                entityIn.setDeltaMovement(0, velocity, 0);
            }
        }
    }

    @Override
    public void fallOn(World worldIn, BlockPos pos, Entity entityIn, float fallDistance) {
        entityIn.fallDistance = 0;
    }

    @Nullable
    @Override
    public ToolType getHarvestTool(BlockState state) {
        return ToolType.PICKAXE;
    }

    @Override
    public int getHarvestLevel(BlockState state) {
        return 2; //Harvest level of 2 means it requires iron or better to harvest
    }

    private boolean isInMiddleOfBlock(BlockPos block, Entity entity) {
        double x = Math.abs(block.getX() - entity.getX());
        double z = Math.abs(block.getZ() - entity.getZ());

        return x > .2d && x < .8d && z > .2d && z < .8d;
    }

}

File: src/main/java/org/manapart/jumpPads/JumpPad.kt
package org.manapart.jumpPads

import net.minecraft.core.BlockPos
import net.minecraft.world.entity.Entity
import net.minecraft.world.level.Level
import net.minecraft.world.level.block.SlimeBlock
import net.minecraft.world.level.block.SoundType
import net.minecraft.world.level.block.state.BlockBehaviour
import net.minecraft.world.level.block.state.BlockState
import net.minecraft.world.level.material.Material
import net.minecraft.world.level.material.MaterialColor
import kotlin.math.abs

private fun createProps(): BlockBehaviour.Properties {
    val padMat = Material.Builder(MaterialColor.COLOR_GREEN).build()
    val props = BlockBehaviour.Properties.of(padMat)
//    props.requiresCorrectToolForDrops()
//    props.harvestTool(ToolType.PICKAXE)
    props.sound(SoundType.SLIME_BLOCK)
    props.strength(4f)
    return props
}

class JumpPad internal constructor(private val velocity: Float) : SlimeBlock(createProps()) {
    override fun stepOn(worldIn: Level, pos: BlockPos, state: BlockState, entityIn: Entity) {
        entityIn.fallDistance = 0f
        super.stepOn(worldIn, pos, state, entityIn)
        if (!entityIn.isCrouching && isInMiddleOfBlock(pos, entityIn)) {
            if (worldIn.isClientSide) {
                entityIn.setDeltaMovement(0.0, velocity.toDouble(), 0.0)
            }
        }
    }

    override fun fallOn(worldIn: Level, state: BlockState, pos: BlockPos, entityIn: Entity, p_154571_: Float) {
        entityIn.fallDistance = 0f
    }

    private fun isInMiddleOfBlock(block: BlockPos, entity: Entity): Boolean {
        val x = abs(block.x - entity.x)
        val z = abs(block.z - entity.z)
        return x > .2 && x < .8 && z > .2 && z < .8
    }

}